src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethods.cs

```diff
@@ -94,6 +94,12 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context)
                         continue;
                     }
 
+                    // Ignore declared type that overrides a base member
+                    if (symbol.IsOverride)
```

Comment:
I am not sure if we want to ignore all overrides. For example, if base type defines a virtual/abstract method and derive type defines an override for it and also defines a new property with a name that violates it, we still want to flag the derive type's property. I think we only want to not flag when both the property and method symbol are override.

Reply:
Looks like the comment is not good enough. Basically if you look at the test you'll see that if you override the property or the method or both no issue is raised. On the other hand if you override one of the 2 and add the complementary member you will have an issue raised.

Reply:
I understand why this works, just that it wasn’t clear from a quick glance. Is there a reason to have your newly added check inside the loop instead of bailing out right at the start of the method?

Reply:
> It’s not the comment but the implementation that doesn’t seem tight enough. I would have expected the newly added if statement condition to check if both symbol.IsOverride and member.IsOverride is true, not just one.

Well if I have both `symbol.IsOverride` and `member.IsOverride` then I will keep reporting on cases where the base class defines both the property and the method and I override only one of them.
I will make some more tests to see if I can somehow find some more restrictive condition to add.

> I understand why your current implementation works, just that it wasn’t clear from a quick glance. Is there a reason to have your newly added check inside the loop instead of bailing out right at the start of the method?

Nope no good reason. I will fix it.

Reply:
Alright so according to this "truth table":

```
local P + nothing => no issue
local P + local M => issue
local P + base M => issue
local P + override M => issue
==> issue when local P + whatever M

override P + nothing => no issue
override P + local M => issue
override P + base M => no issue
override P + override M => no issue
==> issue when override P + local M

local M + nothing => no issue
local M + local P => issue
local M + base P => issue
local M + override P => issue
==> issue when local M + whatever P

override M + nothing => no issue
override M + local P => issue
override M + base P => no issue
override M + override P => no issue
==> issue when override M + local M
```

and based on the fact that the analyzer will kick-in for both the method and the property (when they are present), we could either keep the if located here update it to `symbol.IsOverride && member.ContainingType.Equals(type)` or we can move it to the beginning of the `AnalyzerSymbol` method. Indeed, when the member is an override, the only case where we want to report is when the other member is declared locally AND so the second check will take care of it.

Let me know what's your preference.

Reply:
Thanks for the detailed explanation. I am fine with either approach, with a weaker preference towards just bailing out up front for symbols that are overrides.

Reply:
I will update to add the second part to the if then :D
Not trying to convince you of going for the other solution but do you think the overhead of the various visibility, loop and member checks are neglectable?

Reply:
Yes, I think performance is not really a concern here. Best to come up with a clean, maintainable solution. I am fine with either of your proposed routes. Thanks!